spec/setup/helpers/perf_helper_spec.rb

```diff
@@ -49,5 +72,429 @@ def options
 
   end
 
+  context '.setup_r10k' do
+
+    let!(:masters) {[{'platform' => Beaker::Platform.new('centos-6.5-x86_64')}]}
+
+    it 'adds the private key, configures ssh, and installs git on each master' do
+
+      allow(subject).to receive(:masters).and_return(masters)
+      expect(subject).to receive(:select_hosts).with({:roles => ['master', 'compile_master']}).and_return(masters)
+
+      expect(subject).to receive(:create_remote_file).with(masters[0], "/root/.ssh/id_rsa", /PRIVATE/ )
+      expect(subject).to receive(:create_remote_file).with(masters[0], "/root/.ssh/config", /StrictHostKeyChecking/ )
+
+      expect(subject).to receive(:on).with(masters[0], "chmod 600 /root/.ssh/id_rsa /root/.ssh/config" )
+      expect(subject).to receive(:install_package).with(masters[0], 'git')
+      subject.setup_r10k
+
+    end
+
+  end
+
+  # TODO: refactor to remove duplication
+  describe '#perf_install_pe' do
+
+    context 'when pe is not a pre-aio version' do
+
+      test_is_pre_aio_version = false
+
+      context 'when we are using meep' do
+
+        # TODO: how should @options / @options[:answers] be handled?
+        before {
+          test_options = { :answers => {} }
+          #subject.instance_variable_set(:@options, { })
+          subject.instance_variable_set(:@options, test_options)
+        }
+
+        test_use_meep = true
+        test_pe_ver = '2017.3'
+        let!(:master) { {'pe_ver' => test_pe_ver} }
+
+        test_updated_options = {:answers=>{"puppet_enterprise::profile::master::r10k_remote"=>"/opt/puppetlabs/server/data/puppetserver/r10k/control-repo", "puppet_enterprise::profile::master::r10k_private_key"=>"/root/.ssh/id_rsa"}}
+
+        it 'includes the dashboard and installs pe' do
+
+          expect(subject).to receive(:is_pre_aio_version?).and_return(test_is_pre_aio_version)
+          expect(subject).to receive(:use_meep?).with(master['pe_ver'] || options['pe_ver']).and_return(test_use_meep)
+          expect(subject).to receive(:master).and_return(master)
+
+          # TODO: @options[:answers] ?
+          # expect(subject.instance_variable_get(:@options)). to eq(test_updated_options)
```

Comment:
Since perf_install_pe is only reading the values of the options and not changing them, I don't think we need to assert on their values.
I wouldn't worry too much about this, since for our scenarios use_meep? should always evaluate to true. In fact, we should probably remove the if/else and always assume we are using meep.

Reply:
Sounds good!